server/src/data/updateApp.js

```diff
@@ -28,6 +28,15 @@ const paramsSchema = joi
     })
     .options({ allowUnknown: true })
 
+const isValidSourceUrl = sourceUrl => {
```

Comment:
This could probably be done through Joi with, see https://joi.dev/api/?v=17.4.0#stringurioptions . Using `['http', 'https'` as URI-schemes.